decidim-proposals/app/commands/decidim/proposals/admin/import_participatory_text.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Proposals
+    module Admin
+      # A command with all the business logic when an admin imports proposals from
+      # a participatory text.
+      class ImportParticipatoryText < Rectify::Command
+        # Public: Initializes the command.
+        #
+        # form - A form object with the params.
+        def initialize(form)
+          @form = form
+        end
+
+        # Executes the command. Broadcasts these events:
+        #
+        # - :ok when everything is valid.
+        # - :invalid if the form wasn't valid and we couldn't proceed.
+        #
+        # Returns nothing.
+        def call
+          @form.valid?
+          return broadcast(:invalid) unless form.valid?
+
+          save_participatory_text(form)
```

Comment:
Wrap these in a transaction

Reply:
Hey, the cost of adding the transaction is zero. I'll do as you please but if you haven't checked the code how do you know that it requires docs?

Reply:
Transaction has been added.

Reply:
Going back with the initial comment: I thought you wouldn't want to partial create parts of the text if it fails, otherwise if it fails and you re-import it you might end with duplicated proposals, right?

Reply:
>  if you haven't checked the code how do you know that it requires docs?

The fact that you need to explain why it requires not using a transaction tips me off. This explanation should be explained in the code too, so other devs don't make a mistake 😄